packages/cli/src/metrics/__tests__/prometheus-metrics.service.test.ts

```diff
@@ -22,103 +23,99 @@ describe('PrometheusMetricsService', () => {
 		endpoints: {
 			metrics: {
 				prefix: 'n8n_',
-				includeDefaultMetrics: true,
-				includeApiEndpoints: true,
-				includeCacheMetrics: true,
-				includeMessageEventBusMetrics: true,
+				includeDefaultMetrics: false,
+				includeApiEndpoints: false,
+				includeCacheMetrics: false,
+				includeMessageEventBusMetrics: false,
 				includeCredentialTypeLabel: false,
 				includeNodeTypeLabel: false,
 				includeWorkflowIdLabel: false,
-				includeApiPathLabel: true,
-				includeApiMethodLabel: true,
-				includeApiStatusCodeLabel: true,
+				includeApiPathLabel: false,
+				includeApiMethodLabel: false,
+				includeApiStatusCodeLabel: false,
+				includeQueueMetrics: false,
 			},
 		},
 	});
 
+	const app = mock<express.Application>();
+	const eventBus = mock<MessageEventBus>();
+	const eventService = mock<EventService>();
+	const prometheusMetricsService = new PrometheusMetricsService(
+		mock(),
+		eventBus,
+		globalConfig,
+		eventService,
+	);
+
+	afterEach(() => {
+		jest.clearAllMocks();
+		prometheusMetricsService.disableAllMetrics();
+	});
+
 	describe('init', () => {
 		it('should set up `n8n_version_info`', async () => {
-			const service = new PrometheusMetricsService(mock(), mock(), globalConfig);
+			await prometheusMetricsService.init(app);
 
-			await service.init(mock<express.Application>());
-
-			expect(promClient.Gauge).toHaveBeenCalledWith({
+			expect(promClient.Gauge).toHaveBeenNthCalledWith(1, {
 				name: 'n8n_version_info',
 				help: 'n8n version info.',
 				labelNames: ['version', 'major', 'minor', 'patch'],
 			});
 		});
 
 		it('should set up default metrics collection with `prom-client`', async () => {
-			const service = new PrometheusMetricsService(mock(), mock(), globalConfig);
-
-			await service.init(mock<express.Application>());
+			prometheusMetricsService.enableMetric('default');
+			await prometheusMetricsService.init(app);
 
 			expect(promClient.collectDefaultMetrics).toHaveBeenCalled();
 		});
 
 		it('should set up `n8n_cache_hits_total`', async () => {
-			config.set('endpoints.metrics.includeCacheMetrics', true);
-			const service = new PrometheusMetricsService(mock(), mock(), globalConfig);
-
-			await service.init(mock<express.Application>());
+			prometheusMetricsService.enableMetric('cache');
+			await prometheusMetricsService.init(app);
 
 			expect(promClient.Counter).toHaveBeenCalledWith({
 				name: 'n8n_cache_hits_total',
 				help: 'Total number of cache hits.',
 				labelNames: ['cache'],
 			});
-			// @ts-expect-error private field
-			expect(service.counters.cacheHitsTotal?.inc).toHaveBeenCalledWith(0);
```

Comment:
Exactly, I removed it because I'm trying to avoid accessing private in tests, and because that initial value is already tested in the integration test.